game/configuration.hh

```diff
@@ -47,25 +51,25 @@ class ConfigItem {
 	std::string const getEnumName() const; ///< Returns the selected enum option's text
 
   private:
-	template <typename T> void updateNumeric(xmlpp::Element& elem, int mode); ///< Used internally for loading XML
+	template <typename T> void updateNumeric(xmlpp::Element& elem, unsigned mode); ///< Used internally for loading XML
 	void verifyType(std::string const& t) const; ///< throws std::logic_error if t != type
 	ConfigItem& incdec(int dir); ///< Increment/decrement by dir steps (must be -1 or 1)
 	std::string m_keyName; ///< The config key in the schema file.
 	std::string m_type;
 	std::string m_shortDesc;
 	std::string m_longDesc;
 
-	typedef std::variant<bool, int, double, std::string, StringList> Value;
+	typedef std::variant<bool, unsigned short, int, float, std::string, StringList> Value;
+// I wanted to allow actual conversion (where possible without changing values erroneously) between i() and ui() but it's not as simple to do as I thought. verifyType() is going bye-bye anyway, so maybe there's a better way to implement it properly with the eventual std::visit implementations.
 	bool isDefaultImpl(Value const& defaultValue) const;
 	Value m_value; ///< The current value
 	Value m_factoryDefaultValue; ///< The value from config schema
 	Value m_defaultValue; ///< The value from config schema or system config
 	std::string m_oldValue; ///< A previous value, as output by getValue().
 	std::vector<std::string> m_enums; ///< Enum value titles
-	std::variant<int, double> m_step, m_min, m_max;
-	std::variant<int, double> m_multiplier;
+	std::variant<int, unsigned short, float> m_step, m_min, m_max, m_multiplier;
 	std::string m_unit;
-	int m_sel;
+	unsigned short m_sel;
```

Comment:
Everywhere else you explicitly use std:int64_t or other types with a guaranteed bitlength, yet here you use short. Looks a bit inconsistent to me.

Reply:
If I do it's a merging mistake. I wanted to use exclusively fixed-length types but @Baklap4 wont't have it, so I changed to all non-fixed-width again, except for a few exceptions.